make_test_images/src/compare_manifests.rs

```diff
@@ -149,129 +146,258 @@ fn gather_manifests(manifest_store: &Reader, manifest_label: &str, labels: &mut
+            let mut new_arr: Vec<Value> = arr.into_iter().map(normalize_json).collect();
+
+            // If this is an array of objects with identifiers, sort it.
+            if !new_arr.is_empty()
+                && new_arr.iter().all(|v| v.is_object())
+                && new_arr.iter().all(|v| get_object_identifier(v).is_some())
+            {
+                new_arr.sort_by_key(get_object_identifier);
+            }
+            Value::Array(new_arr)
+        }
+        Value::String(s) => {
+            if GUID_RE.is_match(&s) {
+                Value::String(GUID_RE.replace_all(&s, "<GUID>").to_string())
+            } else if JUMBF_GUID_RE.is_match(&s) {
+                // Only replace the GUID portion, keep the rest of the string
+                let replaced = JUMBF_GUID_RE.replace(&s, |caps: &regex::Captures| {
+                    let prefix = caps.get(0).map_or("", |m| m.as_str());
+                    let guid_re = Regex::new(r"urn:uuid:[0-9a-fA-F\-]{36}").unwrap();
+                    guid_re.replace_all(prefix, "<GUID>").to_string()
+                });
+                Value::String(replaced.to_string())
+            } else if UTC_TIMESTAMP_RE.is_match(&s) {
+                Value::String("<TIMESTAMP>".to_string())
+            } else if VERSION_RE.is_match(&s) {
+                Value::String("<VERSION>".to_string())
+            } else {
+                Value::String(s)
             }
+        }
+        _ => value,
+    }
+}
 
-            for (key, value) in map2 {
-                if map1.get(key).is_none() {
-                    issues.push(format!("Added {path}.{key}: {value}"));
+/// Recursively compare two JSON values
+fn compare_json_values(path: &str, val1: &Value, val2: &Value, issues: &mut Vec<String>) {
+    if val1 == val2 {
+        return;
+    }
+
+    // Suppress reporting changes in hash fields
+    if path.ends_with(".hash") || path.contains("].hash") {
+        return;
+    }
+
+    match (val1, val2) {
+        (Value::Object(map1), Value::Object(map2)) => {
+            let keys1: Vec<_> = map1.keys().collect();
+            let keys2: Vec<_> = map2.keys().collect();
+
+            for key in keys1 {
+                let new_path = format!("{path}.{key}");
+                match map2.get(key) {
+                    Some(v2) => compare_json_values(&new_path, map1.get(key).unwrap(), v2, issues),
+                    None => issues.push(format!("Removed {new_path}: {}", map1.get(key).unwrap())),
                 }
             }
-        }
-        (serde_json::Value::Array(arr1), serde_json::Value::Array(arr2)) => {
-            for (i, (val1, val2)) in arr1.iter().zip(arr2.iter()).enumerate() {
-                compare_json_values(&format!("{path}[{i}]"), val1, val2, issues);
+
+            for key in keys2 {
+                if !map1.contains_key(key) {
+                    issues.push(format!("Added {path}.{key}: {}", map2.get(key).unwrap()));
+                }
             }
```

Comment:
It might be better with something like:
```rust
for (key, value) in map1 {
    // todo
}
```
and same with keys2 so that it doesn't need to be collected, nor the value fetched from the map and unwrapped.